src/terminal/kitty.rs

```diff
@@ -0,0 +1,256 @@
+// Copyright 2018-2019 Sebastian Wiesner <sebastian@swsnr.de>
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//  http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! The kitty terminal.
+//!
+//! kitty is a fast, featureful, GPU based terminal emulator.
+//!
+//! See <https://sw.kovidgoyal.net/kitty/> for more information.
+
+use super::magic;
+use super::url::read_url;
+use failure::Error;
+use image::{ColorType, FilterType};
+use image::{DynamicImage, GenericImageView};
+use std::io::Write;
+use std::process::{Command, Stdio};
+use std::str;
+use url::Url;
+
+/// Whether we run in Kitty or not.
+pub fn is_kitty() -> bool {
+    std::env::var("TERM")
+        .map(|value| value == "xterm-kitty")
+        .unwrap_or(false)
+}
+
+/// Retrieve the terminal size in pixels by calling the command-line tool `kitty`.
+///
+///     kitty +kitten icat --print-window-size
+///
+/// We cannot use the terminal size information from Context.output.size, because
+/// the size information are in columns / rows instead of pixel.
+fn get_terminal_size_in_pixels() -> std::io::Result<(u32, u32)> {
+    use std::io::{Error, ErrorKind};
+
+    let process = Command::new("kitty")
+        .arg("+kitten")
+        .arg("icat")
+        .arg("--print-window-size")
+        .stdout(Stdio::piped())
+        .spawn()?;
```

Comment:
As far as I understand we're doing this for every image?  Let's make a mental note for another refactoring here:  Move this to the entry point of `lib.rs` or even `main.rs` and compute the window size only once, then pass it down in `Context` :)

Reply:
Your assumption is correct - we need to find a way how to integrate the terminal dimension optionally, as if the terminal supports it.